### DISCUSSION TITLE
Go Migrations - Unable to create relation between users collection and X collection

### DISCUSSION THREAD

**Discussion by CRBroughton** [pocketbase/pocketbase#3290] [Q&A]
Hi there, I'm having some trouble creating a relationship between the default users table and a table I'd like to create using Go.

The problem seems to be related to the Id of the users table, however I'm not sure how to fix this issue; I have attempted both to use the '_pb_users_auth_' string, as I saw that being used in the JS migrations, and I have also attempted :

```
usersCollection, err := dao.FindCollectionByNameOrId("users")
```

and then `CollectionId: userCollection.Id` however this approach failed as well. Below is my current migration file:

```
package migrations

import (
	"github.com/pocketbase/dbx"
	"github.com/pocketbase/pocketbase/daos"
	m "github.com/pocketbase/pocketbase/migrations"
	"github.com/pocketbase/pocketbase/models"
	"github.com/pocketbase/pocketbase/models/schema"
	"github.com/pocketbase/pocketbase/tools/types"
)

func init() {
	m.Register(func(db dbx.Builder) error {
		dao := daos.New(db)

		ownerOnly := types.Pointer("@request.auth.id != '' && owners.id ?= @request.auth.id")

		collection := &models.Collection{
			Name:       "items",
			Type:       models.CollectionTypeBase,
			ListRule:   ownerOnly,
			ViewRule:   ownerOnly,
			CreateRule: types.Pointer(""),
			UpdateRule: ownerOnly,
			DeleteRule: ownerOnly,
			System:     false,
			Schema: schema.NewSchema(
				&schema.SchemaField{
					Name:     "name",
					Type:     schema.FieldTypeText,
					Required: true,
					System:   false,
				},
				&schema.SchemaField{
					Name:     "date",
					Type:     schema.FieldTypeDate,
					Required: true,
					System:   false,
				},
				&schema.SchemaField{
					Name:     "owner",
					Type:     schema.FieldTypeRelation,
					Required: true,
					System:   false,
					Options: &schema.RelationOptions{
						CollectionId:  "_pb_users_auth_",
						CascadeDelete: false,
						MinSelect:     types.Pointer(1),
						MaxSelect:     types.Pointer(1),
					},
				},

			),
		}

		err := dao.SaveCollection(collection)

		if err != nil {
			return err
		}
		return nil
	}, func(db dbx.Builder) error {
		dao := daos.New(db)

		collection, err := dao.FindCollectionByNameOrId("items")
		if err != nil {
			return err
		}

		err = dao.DeleteCollection(collection)
		if err != nil {
			return err
		}

		return nil
	})
}
```

Any help is greatly appreciated; I can confirm the migration is being run as I can see the new collection in the GUI, however the relation doesn't see the users table:

![Screenshot from 2023-09-10 15-59-24](https://github.com/pocketbase/pocketbase/assets/71838537/ef877a67-ce1e-458f-963c-3faf42f7389e)

**Comment by ganigeorgiev** [comment]
I've just tested it locally with v0.18.3 and it works fine for me.

What version of PocketBase do you use?

Could you paste your current collections configuration as JSON (you can export it from the _Admin UI > Settings > Export collections_)?

Is it possible that you've executed the migration with incorrect `CollectionId`?
Note that migrations are executed only once and you need to run `migrate down` to rollback it first in order to be able to rerun it.

**Comment by CRBroughton** [comment]
Hi there, thank you for the reply :) I was running V0.18 , however upgrading to V0.18.3 seems to have fixed the issue, not sure why though

![image](https://github.com/pocketbase/pocketbase/assets/71838537/f8d914e3-25ff-4de2-88bc-f22e1d73cb90)

I guess we can consider this resolved :D Thank you for your reply and help.

**Reply by ganigeorgiev** [reply]
There are no changes to the db structure between v0.18.0 and v0.18.3. It could have been a browser cache, I'm not sure.

**Comment by whmeitzler** [comment]
You're specifying the collection name rather than the collection id.

If you want migrations to work, you need to specify a collection Id (you can do this by declaring a BaseCollection with the ID) in the referred collection and use that id in the referring fields relation options. 

This flow is a bit sticky right now, as setting collection ids in a go migration is not very intuitive. The generated migrations sidestep the issue by parsing JSON, which works but doesn't give great hints. I find myself translating JSON collections to Go when working with go migrations 

Id love to hear @ganigeorgiev's thoughts on how we handle go migrations.


EDIT: I missed that you had already tried to get the id from the name. Never mind...

**Reply by ganigeorgiev** [reply]
`_pb_users_auth_` is the id of the default `users` collection.

I don't understand what do you mean by _"This flow is a bit sticky right now, as setting collection ids in a go migration is not very intuitive."_

To create collections programmatically you can also check https://pocketbase.io/docs/go-collections/#create-new-collection-without-data-validations.
